CMakeLists.txt

```diff
@@ -0,0 +1,120 @@
+execute_process(COMMAND ${Python_EXECUTABLE} ${CMAKE_CURRENT_LIST_DIR}/scripts/conan_installer.py RESULT_VARIABLE RTC)
+if (NOT ${RTC} EQUAL 0)
+	message(FATAL_ERROR "Local installer failed")
+endif()
+
+# Dependency config
+conan_cmake_configure(
+	REQUIRES "mtap/0.2" "nlohmann_json/3.10.5"
+	GENERATORS cmake_find_package
+)
+
+# Setup packages using conan-cmake
+message("Generator is multi-config: ${GENERATOR_IS_MULTI_CONFIG}")
+if(GENERATOR_IS_MULTI_CONFIG)
+	foreach (TYPE ${CMAKE_CONFIGURATION_TYPES})
+		conan_cmake_autodetect(SM64_TASFW_SETTINGS BUILD_TYPE ${TYPE})
+		conan_cmake_install(PATH_OR_REFERENCE .
+			BUILD missing
+			SETTINGS ${SM64_TASFW_SETTINGS}
+			SETTINGS "compiler.cppstd=20"
+			INSTALL_FOLDER "${CMAKE_BINARY_DIR}/CMakeFiles/conan_deps"
+		)
+	endforeach()
+else()
+	conan_cmake_autodetect(SM64_TASFW_SETTINGS)
+	conan_cmake_install(PATH_OR_REFERENCE .
+		BUILD missing
+		SETTINGS ${SM64_TASFW_SETTINGS}
+		SETTINGS "compiler.cppstd=20"
+		INSTALL_FOLDER "${CMAKE_BINARY_DIR}/CMakeFiles/conan_deps"
+	)
+endif()
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_BINARY_DIR}/CMakeFiles/conan_deps")
+list(APPEND CMAKE_PREFIX_PATH "${CMAKE_BINARY_DIR}/CMakeFiles/conan_deps")
+
+# Import Conan packages
+foreach(PACKAGE IN ITEMS nlohmann_json mtap)
+	find_package(${PACKAGE} REQUIRED)
+endforeach()
+
+# TODO: separate library files into a separate target
+add_executable(main
+	"sm64-tas-scripting/BitFsConfig.cpp"
+	"sm64-tas-scripting/BitFsConfig.hpp"
+	"sm64-tas-scripting/BitFsPyramidOscillation.cpp"
+	"sm64-tas-scripting/BitFsPyramidOscillation_Iteration.cpp"
+	"sm64-tas-scripting/BitFsPyramidOscillation_RunDownhill.cpp"
+	"sm64-tas-scripting/BitFsPyramidOscillation_TurnAroundAndRunDownhill.cpp"
+	"sm64-tas-scripting/BitFsPyramidOscillation_TurnThenRunDownhill_AtAngle.cpp"
+	"sm64-tas-scripting/BitFsPyramidOscillation_TurnThenRunDownhill.cpp"
+	"sm64-tas-scripting/BrakeToIdle.cpp"
+	"sm64-tas-scripting/Camera.hpp"
+	"sm64-tas-scripting/Game.cpp"
+	"sm64-tas-scripting/Game.hpp"
+	"sm64-tas-scripting/GetMinimumDownhillWalkingAngle.cpp"
+	"sm64-tas-scripting/Inputs.cpp"
+	"sm64-tas-scripting/Inputs.hpp"
+	"sm64-tas-scripting/main.cpp"
+	"sm64-tas-scripting/math.cpp"
+	"sm64-tas-scripting/math.hpp"
+	"sm64-tas-scripting/ObjectFields.hpp"
+	"sm64-tas-scripting/pyramid.cpp"
+	"sm64-tas-scripting/pyramid.hpp"
+	"sm64-tas-scripting/Script_BitFsPyramidOscillation.hpp"
+	"sm64-tas-scripting/Script.cpp"
+	"sm64-tas-scripting/Script_General.hpp"
+	"sm64-tas-scripting/Script.hpp"
+	"sm64-tas-scripting/SharedLib.cpp"
+	"sm64-tas-scripting/SharedLib.hpp"
+	"sm64-tas-scripting/Sm64.hpp"
+	"sm64-tas-scripting/surface.cpp"
+	"sm64-tas-scripting/surface.hpp"
+	"sm64-tas-scripting/Trig.cpp"
+	"sm64-tas-scripting/Trig.hpp"
+	"sm64-tas-scripting/TryHackedWalkOutOfBounds.cpp"
+	"sm64-tas-scripting/Types.hpp"
+	"sm64-tas-scripting/UltraTypes.hpp"
+)
+target_compile_features(main PUBLIC cxx_std_20)
+target_link_libraries(main nlohmann_json::nlohmann_json mtap::mtap)
```

Comment:
target_link_libraries(main nlohmann_json::nlohmann_json mtap::mtap ${CMAKE_DL_LIBS})

Reply:
This was needed to fix it for me